fabrics.c

```diff
@@ -560,9 +559,11 @@ int nvmf_discover(const char *desc, int argc, char **argv, bool connect)
 	set_discovery_kato(&cfg);
 
 	if (!hostnqn)
-		hostnqn = hnqn = nvmf_hostnqn_from_file();
+		hostnqn = nvmf_hostnqn_from_file();
+	if (!hostnqn)
+		hostnqn = nvmf_hostnqn_generate();
 	if (!hostid)
-		hostid = hid = nvmf_hostid_from_file();
+		hostid = nvmf_hostid_from_file();
```

Comment:
Watch out. This patch is actually required, as it differentiate between 'hostid' being a command-line argument (which is static, and can't be freed) and read or generated (which is dynamic, and needs to be freed).

Reply:
`nvmf_hostid_from_file()` calls `nvmf_read_file()` which returns a new string via `strndup()`. I can't see where we return a static string (which would be a very confusing API...)